doc/acl/Extend-L3V6ACLs.md

```diff
@@ -0,0 +1,308 @@
+      - [Option-A: Follow the _existing_ optimization for Mirror ACL table](#option-a-follow-the-existing-optimization-for-mirror-acl-table)
+        - [Pros](#pros)
+        - [Cons](#cons)
+      - [Option-B: Include IPv4 match fields in table type L3V6](#option-b-include-ipv4-match-fields-in-table-type-l3v6)
+        - [Pros](#pros-1)
+        - [Cons](#cons-1)
+      - [Option-C: Create a new ACL Table Type L3V4V6](#option-c-create-a-new-acl-table-type-l3v4v6)
+        - [Pros and Cons](#pros-and-cons)
+      - [Implementation](#implementation)
+        - [Orchagent](#orchagent)
+        - [STATE\_DB](#state_db)
+    - [SAI API](#sai-api)
+    - [Configuration and management](#configuration-and-management)
+    - [Warmboot and Fastboot Design Impact](#warmboot-and-fastboot-design-impact)
+    - [Restrictions/Limitations](#restrictionslimitations)
+    - [Testing Requirements/Design](#testing-requirementsdesign)
+      - [Unit Test cases](#unit-test-cases)
+      - [System Test cases](#system-test-cases)
+
+### Revision  
+
+| Rev | Date     | Author          | Change Description |
+|:---:|:--------:|:---------------:|--------------------|
+| 0.1 | 18/Feb/23 | Ravindranath (**Marvell**)   | Initial Version.  |
+
+### Scope  
+
+
+This document provides the high level design in SONiC to combine L3 and L3V6 ACL tables in SAI on supported hardware.
+
+### Terminology
+
+
+
+| SONiC User ACL Table Types     | Description |
+|---------------|--------------------|
+| __L3__ ACL table type | A built-in ACL table type in SONiC that can match __IPv4__ fields along with incoming port and support ACL redirect and packet ACL actions.|
+| __L3V6__ ACL  table type | A built-in ACL table type in SONiC that can match __IPv6__ fields along with incoming port  and support redirect and packet ACL actions.|
+| __Mirror__ ACL  table type | A built-in ACL table type in SONiC that can match __IPv4__ fields and supports __Mirror__ ACL action. |
+| __MirrorV6__ ACL  table type | A built-in ACL table type in SONiC that can match __IPv6__ fields and supports __Mirror__ ACL action. |
+| __Custom__ ACL  table types | A recent enhancement where one or more ACL table types can be created by the user specifying the match-field list, action-list and the bindpoint-types list.|
+
+
+### Overview 
+
+This document describes the orchagent support by which user created L3 ACL table and L3V6 ACL tables are combined into a single SAI/ASIC ACL table on platforms optimized for this feature.
+
+In several ASICs,  IPv4 and IPv6 ACL rules can be supported in the same ACL table. Further, in many hardware, when the hardware tables (TCAMs) are configured to match IPv6 addresses, the hardware can use the same resources to match the corresponding IPv4 packet fields without incurring additional hardware resources. For example, IPv6 Destination address (128b) and IPv4 Destination address (32b) keys can be fit using only 128 bits instead of 128 + 32bits.
+
```

Comment:
Can you share the changes required in the existing ACL_TABLE and ACL_RULE tables in config-db to support V4V6 ACL optimization?

Reply:
As per the feedback received during the community discussion, we are creating a new ACL table type. We have given an example configuration to use the new ACL table type in the "Configuration and Management" section.